pkg/operator/cloudformation/ensure.go

```diff
@@ -79,13 +60,37 @@ func (c *CloudFormation) Ensure() error {
 
 	// TODO
 	//
-	//     add inhibition operator function to cancel reconciliation if stack is updating
 	//     emit deployment event via Specta, if updated
 	//
 
 	return nil
 }
 
+func (c *CloudFormation) temPar(rel []cache.Object) []types.Parameter {
+	var par []types.Parameter
+
+	for k, v := range c.env.CloudformationParameters {
+		par = append(par, types.Parameter{
+			ParameterKey:   aws.String(k),
+			ParameterValue: aws.String(v),
+		})
+	}
+
+	// Inject all desired artifact versions into the parameters that we are just
+	// about to deploy. Injecting those parameters after all user inputs have been
+	// applied above guarantees that only the release versions as defined in the
+	// release source repository will ever be applied.
+
+	for _, x := range rel {
+		par = append(par, types.Parameter{
+			ParameterKey:   aws.String(x.Parameter()),
+			ParameterValue: aws.String(x.Version()),
```

Comment:
Using the conditional version here instead of the static desired state fixes the bug mentioned.